src/terra-dev-site/app/demo/EventEmitter.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import EventEmitter from 'terra-application/lib/utils/event-emitter';
+
+class EventEmitterExample extends React.Component {
+  constructor() {
+    super();
+
+    this.state = {
+      counter: 0,
+    };
+
+    this.listener = this.listener.bind(this);
+    this.eventEmitter = new EventEmitter();
```

Comment:
I'm on the fence. If it was me developing an app, I'd do my best to create a functional component, but the example does seem a bit less clear.... I think i lean towards implementing this as a functional component and using hooks. Other thoughts?

Reply:
I thought about using a functional component with useEffect in this example similar to [this](https://reactjs.org/docs/hooks-effect.html#explanation-why-effects-run-on-each-update). I didn't go with that because this should only remove all listeners once when unmounting instead of after every render. There are certainly use cases where listeners may need to be added and removed as necessary after every render. This is a simple example illustrating the importance to remove all listeners when done.

Reply:
the `useEffect` above will only re-trigger if the `eventEmitter` changes, signified by the array argument at the end 
```
}, [eventEmitter]);
```

not a big deal either way

Reply:
@yuderekyu  Ahh! I didn't know about the optimization option with the additional argument. I switched it and it works great. I can certainly change it to functional if others don't mind about it being less clear.

Reply:
#teamHooks

Reply:
I've changed it to a function component using hooks.